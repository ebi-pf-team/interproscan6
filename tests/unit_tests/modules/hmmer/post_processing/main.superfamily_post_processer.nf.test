nextflow_process {

    name "Test Process SUPERFAMILY_POST_PROCESSER"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "SUPERFAMILY_POST_PROCESSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"
        symlink "tests/unit_tests/test_inputs/hmmscan_parser/hmmscan.out"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/selfhits"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/cla"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/model_tab"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/pdbj95d"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmscan_parser/hmmscan.out")
                input[1] = [
                    "meta/bin/superfamily/ass3.pl",
                    "meta/tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/selfhits",
                    "meta/tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/cla",
                    "meta/tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/model_tab",
                    "meta/tests/unit_tests/test_inputs/hmmer_post_processing/superfamily/pdbj95d",
                    "binary-switches"
                ]
                input[2] = file("tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa")
                input[3] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
