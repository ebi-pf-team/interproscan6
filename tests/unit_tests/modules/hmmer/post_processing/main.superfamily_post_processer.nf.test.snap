{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmscan.out_ass3_output:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T09:27:20.037785"
    }
}